editor/plugins/area_2d_editor_plugin.h

```diff
@@ -0,0 +1,71 @@
+/**************************************************************************/
+/*  area_2d_editor_plugin.h                                               */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef AREA_2D_EDITOR_PLUGIN_H
+#define AREA_2D_EDITOR_PLUGIN_H
+
+#include "editor/editor_plugin.h"
+
+class Area2D;
+class CanvasItemEditor;
+
+class Area2DEditorPlugin : public EditorPlugin {
+	GDCLASS(Area2DEditorPlugin, EditorPlugin)
+
+	StringName grav_vec_pname = StringName("gravity_point_center");
+	StringName grav_ispt_pname = StringName("gravity_point");
+	StringName grav_dist_pname = StringName("gravity_point_unit_distance");
+
+	CanvasItemEditor *canvas_item_editor = nullptr;
+	Area2D *node = nullptr;
+
+	enum Action {
+		ACTION_NONE,
+		ACTION_MOVING_POINT,
+	};
+
+	Action action;
```

Comment:
```suggestion
	Action action = ACTION_NONE;
```
Initialize all primitive types